internal/armada/configuration/types.go

```diff
@@ -80,12 +85,100 @@ type PulsarConfig struct {
+	DefaultJobTolerationsByResourceRequest map[string][]v1.Toleration
+	// Pods of size greater than this are rejected at submission.
+	MaxPodSpecSizeBytes uint
+	// Jobs requesting less than this amount of resources are rejected at submission.
+	MinJobResources v1.ResourceList
+	// Default value of GangNodeUniformityLabelAnnotation if not set on submitted jobs.
+	// TODO(albin): We should add a label to nodes in the nodeDb indicating which cluster it came from.
+	//              If we do, we can default to that label if the uniformity label is empty.
+	DefaultGangNodeUniformityLabel string
+	// Minimum allowed termination grace period for pods submitted to Armada.
+	// Should normally be set to a positive value, e.g., "10m".
+	// Since a zero grace period causes Kubernetes to force delete pods, which may causes issues with container resource cleanup.
+	//
+	// The grace period of pods that either
+	// - do not set a grace period, or
+	// - explicitly set a grace period of 0 seconds,
+	// is automatically set to MinTerminationGracePeriod.
+	MinTerminationGracePeriod time.Duration
+	// Max allowed grace period.
+	// Should normally not be set greater than single-digit minutes,
+	// since cancellation and preemption may need to wait for this amount of time.
+	MaxTerminationGracePeriod time.Duration
+	// Default activeDeadline for all pods that don't explicitly set activeDeadlineSeconds.
+	// Is trumped by DefaultActiveDeadlineByResourceRequest.
+	DefaultActiveDeadline time.Duration
+	// Default activeDeadline for pods with at least one container requesting a given resource.
+	// For example, if
+	// DefaultActiveDeadlineByResourceRequest: map[string]time.Duration{"gpu": time.Second},
+	// then all pods requesting a non-zero amount of gpu and don't explicitly set activeDeadlineSeconds
+	// will have activeDeadlineSeconds set to 1.
+	// Trumps DefaultActiveDeadline.
+	DefaultActiveDeadlineByResourceRequest map[string]time.Duration
+}
+
+// SchedulingConfig contains config controlling the Armada scheduler.
+//
+// The Armada scheduler is in charge of assigning pods to cluster and nodes.
+// The Armada scheduler is part of the Armada control plane.
+//
+// Features:
+// 1. Queuing and fairly dividing resources between users.
+// 2. Fair preemption, including between jobs of equal priority to balance resource allocation.
+// 3. Gang scheduling, optional across clusters, and with lower and upper bounds on the number of jobs scheduled.
+//
+// Note that Armada still relies on kube-scheduler for binding of pods to nodes.
+// This is achieved by adding to each pod created by Armada a node selector that matches only the intended node.
 type SchedulingConfig struct {
 	// Set to true to disable scheduling
 	DisableScheduling bool
 	// Set to true to enable scheduler assertions. This results in some performance loss.
 	EnableAssertions bool
-	Preemption       PreemptionConfig
```

Comment:
Preemption is so integrated it doesn't make sense to keep the config in a separate struct anymore.